parsers/CA_AB.py

```diff
@@ -79,9 +79,21 @@ def fetch_price(
     prices = PriceList(logger)
     for row in csv.reader(response.text.split("\r\n\r\n")[2].splitlines()[1:]):
         if row[1] != "-":
+            try:
+                date_time = datetime.strptime(row[0], "%m/%d/%Y %H").replace(
+                    tzinfo=TIMEZONE
+                )
+            except ValueError:
+                # Handle special case where hour is `24`
+                row[0] = row[0].replace(" 24", " 00")
+                date_time = datetime.strptime(row[0], "%m/%d/%Y %H").replace(
+                    tzinfo=TIMEZONE
+                )
+                date_time += timedelta(days=1)
```

Comment:
I think it would be easier to split the date string into two parts, the date and the time.
Once we have the time seperate we can parse it as an int and do:
```python
datetime.strptime(f"{date} {hour - 1}", "%m/%d/%Y %-H")
```

This would remove the need for the seccond special operation and would give us an accurate index for all hours. If hour 24 is the next day we can simply shift back the hours by 1 using `+ timedelta(hours=1)` as that is fully datetime aware.

Reply:
this implementation is better actually see latest commit